src/App.tsx

```diff
@@ -31,65 +31,104 @@ export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  let visibleGoods = [...goods];
+
+  if (sortType === 1 && isReversed === false) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return a.localeCompare(b);
+    });
+  } else if (sortType === 2 && isReversed === false) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return a.length - b.length;
+    });
+  } else if (sortType === 1 && isReversed === true) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return b.localeCompare(a);
+    });
+  } else if (sortType === 2 && isReversed === true) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return a.length - b.length;
+    }).reverse();
+  } else if (sortType === 0 && isReversed === true) {
+    visibleGoods = visibleGoods.reverse();
+  }
```

Comment:
you can also use a switch case statement and finally use an if statement for isReversed
something like that:
``` switch (sortType) {
    case SortType.ALPHABET:
      visibleGoods.sort((a, b) => a.localeCompare(b));
      break;
    case ...
  }

  if (isReversed) {
    visibleGoods.reverse();
  }

Reply:
'if' was used as the example at the current topic on mate. Also I'm using && at almost all of them so using 'case' could be confusing.